### ISSUE TITLE
MCMC backends discard the per-draw divergence mask, keeping only the count

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#2431]
The MCMC backends compute a per-draw divergence mask and then discard it,
keeping only the count:

```python
n_divergent = int(jnp.sum(divergent))     # backends/mcmc/ghmc.py:383
...
diagnostics = {..., "n_divergent": n_divergent, ...}
```

`info.is_divergent` is already threaded through `_sample_scan`
(`fitter.py:4643` and neighbours), so the array is in hand at the moment it is
summed.

Consequence: "which draws diverged, and where in parameter space" is not
answerable after a fit, from any saved artifact. That is the first question
worth asking about a divergent chain, because a divergence count alone cannot
distinguish a hard prior boundary from a correlated ridge from a stiff
likelihood — and those want different fixes.

Concretely, this blocked a real diagnosis. A grid cell showed 14 divergences in
1200 draws with R-hat 1.0037 and 36% of iterations at the tree-depth cap. The
natural first test — do the divergent transitions concentrate near a parameter's
prior bound — could not be run, because the saved npz holds 1200 draws and no
mask and the json holds only the count. The analysis had to proceed indirectly,
through the pushforward geometry of the whole chain, which answers a related
question rather than that one.

Suggested: keep the mask in `diagnostics` under a per-draw key, e.g.
`"divergent"` as a boolean array of shape `(n_chains, n_samples)` alongside the
existing `"n_divergent"` scalar. Consumers that only want the count are
unaffected. It is ~1 bit per draw before compression, so a 4 x 300 fit costs
1200 booleans.

If the memory is unwelcome by default, an opt-in flag on the sampler call would
still make the diagnosis possible when someone is looking for it, which is
better than it being impossible.
